### PR TITLE
Wallet API doc update

### PR DESCRIPTION
**Pull request by VoidingWarranties**
Not done yet, needs clarification regarding transaction JSON.

### CODE REVIEW

File: doc/API.md [hunk 1]
```diff
@@ -1000,341 +1013,333 @@ Response: standard.
...
+        // TODO: fill this out
+      ],
+      "siafundinputs": [
+        // TODO: fill this out
+      ],
+      "siafundoutputs": [
+        // TODO: fill this out
+      ],
+      "minerfees": [
+        // TODO: fill this out, why is this an array of types.Currency's?
```
Comment (VoidingWarranties) [inline doc/API.md]: why is this an array of types.Currency's?

File: doc/API.md [hunk 2]
```diff
@@ -1000,341 +1013,333 @@ Response: standard.
...
+      ],
+      "arbitrarydata": [
+        // TODO: is there a restriction on arbitrary data? Why is it an array of array of bytes vs a single byte array?
```
Comment (VoidingWarranties) [inline doc/API.md]: is there a restriction on arbitrary data? Why is it an array of array of bytes vs a single byte array?

**Comment by VoidingWarranties** [comment]
In particular I had a difficult time updating the response schema for `/wallet/transaction/:id` from our old style, specified in go, to our new style, with JSON examples.

**Comment by DavidVorick** [comment]
What is the status on this?

File: doc/api/Wallet.md [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+{
+  // All of the confirmed transactions appearing between height 'startheight'
+  // and height 'endheight' (inclusive).
+  "confirmedtransactions": [
+    {
+      // See the documentation for '/wallet/transaction/:id' for more information.
+    }
+  ],
+
+  // All of the unconfirmed transactions.
```
Comment (VoidingWarranties) [inline doc/api/Wallet.md]: Between 'startheight and 'endheight' or all of them?

**Comment by VoidingWarranties** [comment]
Other than the one comment I just made it should be good to go pending review.

File: doc/api/Wallet.md (hunk 3 above)
Comment (DavidVorick) [inline doc/api/Wallet.md]: unconfirmed transactions don't have a height, therefore all of them.

**Pull request merged**